altair/vegalite/v5/api.py

```diff
@@ -3816,33 +3816,140 @@ def add_selection(self, *params) -> Self:  # noqa: ANN002
-            unique among all parameters within the chart.
+            unique among all parameters within the chart
         bind_x : boolean, default True
-            If true, then bind the interactive scales to the x-axis
+            Bind the interactive scales to the x-axis
         bind_y : boolean, default True
-            If true, then bind the interactive scales to the y-axis
+            Bind the interactive scales to the y-axis
+        tooltip : boolean, default True,
+            Add a tooltip containing the encodings used in the chart
+        legend : boolean or string, default True
+            A single encoding channel to be used to create a clickable legend.
+            The deafult is to guess from the spec based on the most commonly used legend encodings.
 
         Returns
         -------
         chart :
-            copy of self, with interactive axes added
+            copy of self, with interactivity added
 
         """
         encodings: list[SingleDefUnitChannel_T] = []
         if bind_x:
             encodings.append("x")
         if bind_y:
             encodings.append("y")
-        return self.add_params(selection_interval(bind="scales", encodings=encodings))
+        interactive_chart = self.add_params(
+            selection_interval(bind="scales", encodings=encodings)
+        ).copy()
+        # We can't simply use configure_mark since configure methods
+        # are not allowed in layered specs
+        if tooltip:
+            if isinstance(interactive_chart.mark, str):
+                interactive_chart.mark = {
+                    "type": interactive_chart.mark,
+                    "tooltip": tooltip,
+                }
+            else:
+                interactive_chart.mark.tooltip = tooltip
+
+        if legend:
+            if isinstance(legend, str):
+                legend_encoding = legend
+            else:
+                # Set the legend to commonly used encodings by default
+                possible_legend_encodings = [
+                    "color",
+                    "fill",
+                    "shape",
+                    "stroke",
+                    "opacity",
+                    "fillOpacity",
+                    "strokeOpacity",
+                    "strokeWidth",
+                    "strokeDash",
+                    "angle",  # TODO Untested
+                    "radius",  # TODO Untested
+                    # "size",  # TODO Currently size is not working, renders empty legend
+                ]
+                legend_encoding = next(
+                    (
+                        enc
+                        for enc in possible_legend_encodings
+                        if not isinstance(
+                            interactive_chart.encoding[enc],
+                            utils.schemapi.UndefinedType,
+                        )
+                    ),
+                    None,  # type: ignore
+                )
```

Comment:
I'm unsure what the best way is to deal with this mypy error:

```
altair/vegalite/v5/api.py:3903: error: Generator has incompatible item type
"str"; expected
"Literal['radius', 'radius2', 'color', 'fill', 'stroke', 'opacity', 'fillOpacity', 'strokeOpacity', 'strokeWidth', 'strokeDash', 'size', 'angle', 'shape']"
 [misc]
                            enc
```

I could ignore it as for the default argument, but maybe there is a better way?

Reply:
```suggestion
                legend_encoding = next(
                    (
                        enc
                        for enc in possible_legend_encodings
                        if not utils.is_undefined(
                            interactive_chart.encoding[enc]
                        )
                    ),
                    None,  # type: ignore
                )
```

Sorry I've hijacked this thread for this one, but I do know how else to do a multi-line suggestion.